src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -117,6 +117,7 @@ def _accumulate_energy_trace(self):
                     self._dissipate_infinite_consumption(trade)
 
     def update_area_kpi(self, area):
+        self.total_energy_demanded_wh = 0
         self._accumulate_total_energy_demanded(area)
```

Comment:
Correct me if I'm wrong `update_area_kpi ` is triggered for every tick meaning that the same_demanded energy would be recalculated for every ticket. I think it could be moved to class constructor as I don't see the need for it to re-calculate an static value for every tick

Reply:
It is updated every market slot actually, not on every tick. Moreover, this was one of the major issues that this PR solves. `_accumulate_total_energy_demanded` is recursive. Therefore, if it is called with the Grid area, it will set the `total_energy_demanded_wh` to 0 at the beginning of every recursion. This will obviously work for single devices, but as I mentioned above, doing this on a neighbourhood or a grid will set this value to the demand of the last load in the recursion.

Reply:
Yes you are right it is indeed called per market_slot BUT if you keep `total_energy_demanded_wh` 
 as part of `event_activate_energy ` then `_accumulate_total_energy_demanded` could be moved to the constructor of this class to save the re-accumulating at every slot.

Reply:
This will have the effect though, that the self-sufficiency and self-consumption will not have correct results during the simulation. I'll redirect you to the second comment chain with @hannesdiedrich , where I describe exactly the reason why it is important IMO to recalculate this value on every market slot (as long as we want the self sufficiency and self consumption values to make sense during the simulation too, and not only at the end of the simulation)

Reply:
Thanks a lot. Convinced :-)